embedding/csharp/Tizen.Flutter.Embedding/FlutterApplication.cs

```diff
@@ -51,6 +51,15 @@ public class FlutterApplication : CoreUIApplication, IPluginRegistry
         /// </summary>
         protected bool IsTopLevel { get; set; } = false;
 
+        /// <summary>
+        /// The renderer type of the engine. Defaults to kEGL. If the profile is wearable, defaults to kEvasGL.
+        /// </summary>
+#if WEARABLE_PROFILE
+        protected FlutterDesktopRendererType RendererType { get; set; } = FlutterDesktopRendererType.kEvasGL;
+#else
+        protected FlutterDesktopRendererType RendererType { get; set; } = FlutterDesktopRendererType.kEGL;
```

Comment:
Avoid declaring the same property twice. You may either move the location of `#if` or assign the value inside the class constructor.

Also the `WEARABLE_PROFILE` macro is not defined.

Reply:
> Avoid declaring the same property twice. You may either move the location of #if or assign the value inside the class constructor.

I will move it to constructor.

> Also the WEARABLE_PROFILE macro is not defined.

https://github.com/flutter-tizen/flutter-tizen/blob/592bb17f6b91eb592485389ba738295e98e7b261/lib/build_targets/package.dart#L364

I thought I could use a macro with this predefine.
Is there any way to get the profile in flutter-tizen?

Reply:
Yeah, the macro is defined by the C++ compiler but not by the .NET compiler. The .NET equivalent is to
- pass the `/p:DefineConstants=WEARABLE_PROFILE` property to the `dotnet build` command as an argument (you need to modify `package.dart`), or
- write the property to the `Runner.csproj.user` file (you need to modify `updateDotnetUserProjectFile` of `tizen_project.dart`).

I'd like to avoid the both but if we really have to choose one, I would prefer the latter.

Reply:
Thank you for your advice.
I tried the second of the ways you suggested, but it didn't work the way I thought it would
I added the below code to tizen_project.dart and tested it, but the WEARABLE_PROFILE define was not found in C#.

```dart
 final Iterable<XmlElement> elementsDefineConstants =
      document.findAllElements('DefineConstants');
  if (elementsDefineConstants.isEmpty) {
    final XmlBuilder builder = XmlBuilder();
    builder.element('PropertyGroup', nest: () {
      builder.element(
        'DefineConstants',
        nest: 'WEARABLE_PROFILE',
      );
    });
    document.rootElement.children.add(builder.buildFragment());
  } else {
    for (final XmlElement element in elementsDefineConstants) {
      element.innerText = 'WEARABLE_PROFILE';
    }
  }
```
Runner.csproj.user
```csproj
<?xml version="1.0" encoding="utf-8"?>
<Project ToolsVersion="Current" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
  <PropertyGroup>
    <FlutterEmbeddingPath>/home/junsu/dev/os/f-project/flutter-tizen/embedding/csharp/Tizen.Flutter.Embedding/Tizen.Flutter.Embedding.csproj</FlutterEmbeddingPath>
  </PropertyGroup>
  <PropertyGroup>
    <DefineConstants>WEARABLE_PROFILE</DefineConstants>
  </PropertyGroup>
</Project>
```

How about adding it to package.dart as the first of your suggestions?
I think this way is better because it can direct access(?) to buildInfo.deviceProfile.
```diff
--- a/lib/build_targets/package.dart
+++ b/lib/build_targets/package.dart
@@ -144,6 +144,7 @@ class DotnetTpk extends TizenPackage {
       if (buildMode.isPrecompiled) 'Release' else 'Debug',
       '-o',
       '${outputDir.path}/', // The trailing '/' is needed.
+      '/p:DefineConstants=${buildInfo.deviceProfile.toUpperCase()}_PROFILE',
       tizenProject.editableDirectory.path,
     ]);
     if (result.exitCode != 0) {

```

Reply:
Alright. Only the first option seems to be feasible.